src/tlo/methods/wasting.py

```diff
@@ -1,41 +1,1277 @@
+        complications, applicable to SAM cases only, requiring inpatient care.
+        :param person_id: individual id
+        :param pop_dataframe: population dataframe
+        """
+        df = pop_dataframe
+        p = self.parameters
+
+        # check if person is not wasted
+        if ((df.at[person_id, 'un_WHZ_category'] == 'WHZ>=-2') &
+                (df.at[person_id, 'un_am_MUAC_category'] == '>=125mm') & (~df.at[person_id, 'un_am_bilateral_oedema'])):
+            df.at[person_id, 'un_clinical_acute_malnutrition'] = 'well'
+
+        # severe acute malnutrition -- MUAC < 115 mm and/or WHZ < -3 and/or bilateral oedema
+        elif ((df.at[person_id, 'un_am_MUAC_category'] == '<115mm') | (df.at[person_id, 'un_WHZ_category'] == 'WHZ<-3')
+              | (df.at[person_id, 'un_am_bilateral_oedema'])):
+            df.at[person_id, 'un_clinical_acute_malnutrition'] = 'SAM'
+            # apply symptoms to all SAM cases
+            self.wasting_clinical_symptoms(person_id=person_id)
+
+        else:
+            df.at[person_id, 'un_clinical_acute_malnutrition'] = 'MAM'
+
+        # Determine if SAM episode has complications
+        if df.at[person_id, 'un_clinical_acute_malnutrition'] == 'SAM':
+            if self.rng.random_sample() < p['prob_complications_in_SAM']:
+                df.at[person_id, 'un_sam_with_complications'] = True
+            else:
+                df.at[person_id, 'un_sam_with_complications'] = False
+        else:
+            df.at[person_id, 'un_sam_with_complications'] = False
+
+        assert not (df.at[person_id, 'un_clinical_acute_malnutrition'] == 'MAM') & \
+                   (df.at[person_id, 'un_sam_with_complications'])
+
+    def date_of_outcome_for_untreated_wasting(self, person_id):
+        """
+        Helper function to use the duration of the wasting episode to get date of outcome (recovery, progression,
+        or death)
+        :param person_id:
+        :return: date of outcome, which can be recovery to no wasting or progression to severe wasting from moderate
+        wasting; or recovery to moderate wasting or death due to severe wasting
+        """
+        df = self.sim.population.props
+        p = self.parameters
+        whz_category = df.at[person_id, 'un_WHZ_category']
+
+        # moderate wasting (for progression to severe, or recovery to no wasting) -----
+        if whz_category == '-3<=WHZ<-2':
+            # Allocate the duration of the moderate wasting episode
+            duration_mod_wasting = int(max(p['min_days_duration_of_wasting'], p['duration_of_untreated_mod_wasting']))
+            # Allocate a date of outcome (progression, or recovery)
+            date_of_outcome = df.at[person_id, 'un_last_wasting_date_of_onset'] + DateOffset(days=duration_mod_wasting)
+            return date_of_outcome
+
+        # severe wasting (for death, or recovery to moderate wasting) -----
```

Comment:
But is there a natural progress to death or recovery from severe wasting? I only found it within the interventions, but I would expect children dying/recovering even if they were not sent for intervention.

Reply:
@andrew-phillips-1 and @timcolbourn, When moderate wasting is initiated, and hence the WHZ changes, the MUAC and presence of oedema are now determined only in these individuals (not in all children under 5). The clinical state of acute malnutrition (well/MAM/SAM) is updated based on these, and if SAM, it is determined whether the individual has complications. **I will now add that if SAM, it is also determined whether the individual will die due to SAM** (i.e., add the death from SAM if not treated). _I just need to make sure these individuals are not on treatment or are not already scheduled for death._ -- **Do you think the death from SAM should be determined every time the WHZ changes and the individual has SAM (after the change), or only if the individual didn't have SAM before the change?** I would vote for the former, as it can happen in many combinations. If they had SAM but WHZ changed for sure, MUAC and oedema may have changed, it could be that one symptom gets better but another worsens, so it might be time to check if they are supposed to die again.

After moderate wasting is initiated, the children are scheduled to progress to severe wasting or recover to normal wasting after the duration of the moderate wasting.

My plan is, similarly to what happens with moderate wasting, when the progress to severe wasting is initiated, and hence the WHZ changes, all the consequent changes are made—MUAC, oedema, clinical state, and if SAM, complications/death. **It will be checked if the person is scheduled to die, and if not, the recovery to moderate wasting will be scheduled after the duration of the severe wasting** (i.e., the natural recovery from severe wasting added).

Does this seem reasonable to you?

Reply:
@timcolbourn and @andrew-phillips-1, I also need to add that once they recover from severe wasting to moderate wasting, they should be scheduled again after the duration of moderate wasting to either progress back to severe wasting or recover to no wasting. Do you agree?

Reply:
Also, the recovery from moderate to normal, or severe to moderate wasting occurs after the duration of moderate or severe wasting (since the onset of that wasting), respectively. However, I’m not sure when we want the death due to SAM to occur (as SAM can occur alongside any wasting), and I would say it could happen at any time since the current wasting incidence.

The death is determined when the clinical state is updated, so we could set it to xx days since then, i.e., assuming an average number of days to death, or we could choose randomly within a 0-yy days range. I will look up some reference(s) we could base this on, but if you know of any useful ones, let me know.

Reply:
Hi @EvaJanouskova  I would have to have time to read up again about wasting before I would feel comfortable in advising on these questions.  If you have done that reading I am happy to defer to you and Tim on the decisions.

Reply:
Hi Eva, that all sounds reasonable yes. Just to note MUAC itself is not really an independent predictor of death, it is used in diagnosis of wasting when scales and height measurement are not available. So the death should be based on the underlying SAM state, whether that is diagnosed by a healthworker (e.g. with MUAC) or not. 

Good idea to look up the average days to death following the start of SAM for those who die, and use that.

Reply:
@timcolbourn, There is actually nothing as official diagnosis by healthworker in the module, there are only the clinical signs the children have - WHZ, MUAC, odema. Based on these the clinical state is determined
SAM: severe wasting (WHZ < -3) and/or MUAC < 115 mm and/or bilateral oedema
MAM: moderate wasting (-3 ≤ WHZ < -2) and/or 115 mm ≤ MUAC < 125 mm

If SAM is being determined as the clinical state, it is stochastically decided on whether the child has complications and then a linear model is used to predict (with predictors being the clinical signs and complications) whether the child will die due to SAM.